reconcile/utils/terrascript_client.py

```diff
@@ -488,7 +488,7 @@ def populate_route53(self, desired_state, default_ttl=300):
             acct_name = zone['account_name']
 
             # Ensure zone is in the state for the given account
-            zone_id = safe_resource_id(f"{zone['name']}")
+            zone_id = safe_resource_id(f"{zone.pop('identifier')}")
```

Comment:
Modifying `zone` inside the f-string - or a read-only function argument for that matter - is going to cause confusion. 
```suggestion
            zone_id = zone.pop('identifier')
            zone_id = safe_resource_id(zone_id)
```
(or whatever if `zone_id` becomes `None`.

Reply:
fair enough. https://github.com/app-sre/qontract-reconcile/pull/2071/commits/779a6a07f3efa1546ac9a004504192229fa674c8